### PR TITLE
Remove special-casing of Synonym filters in AnalysisRegistry

### PR DESCRIPTION
**Pull request by romseygeek**
The synonym filters no longer need access to the AnalysisRegistry in their
constructors, so we can remove the special-case code and move them to the
common analysis module

### CODE REVIEW

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymGraphTokenFilterFactory.java [hunk 1]
```diff
@@ -26,16 +26,18 @@
...
-import java.io.IOException;
 import java.util.List;
 import java.util.function.Function;
 
 public class SynonymGraphTokenFilterFactory extends SynonymTokenFilterFactory {
 
-    public SynonymGraphTokenFilterFactory(IndexSettings indexSettings, Environment env, AnalysisRegistry analysisRegistry,
-                                     String name, Settings settings) throws IOException {
-        super(indexSettings, env, analysisRegistry, name, settings);
+    public SynonymGraphTokenFilterFactory(IndexSettings indexSettings, Environment env,
```
Comment (martijnvg) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymGraphTokenFilterFactory.java]: I think this constructor can now be package protected?

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java [hunk 2]
```diff
@@ -41,8 +46,8 @@
     protected final Settings settings;
     protected final Environment environment;
 
-    public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, AnalysisRegistry analysisRegistry,
-                                      String name, Settings settings) throws IOException {
+    public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env,
```
Comment (martijnvg) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java]: I think this constructor can now be package protected too?

**Review by martijnvg — APPROVED** [review]
LGTM - Thanks for moving these token filters, that was not easy.

**Comment by romseygeek** [comment]
please retest this

**Comment by romseygeek** [comment]
In 6.x the Synonym filter factories still require AnalysisRegistry for backwards compatibility so this can't be backported

**Pull request merged**